src/server/app.js

```diff
@@ -0,0 +1,70 @@
+var express = require('express');
+var path = require('path');
+var favicon = require('serve-favicon');
+var requestLogger = require('./lib/requestLogger');
+var cookieParser = require('cookie-parser');
+var bodyParser = require('body-parser');
+var compression = require('compression');
+var config = require('./config');
+
+var routes = require('./routes/index');
+var proxy = require('./routes/proxy');
+
+var app = express();
+
+// view engine setup
+app.set('views', path.join(__dirname, 'views'));
+app.set('view engine', 'jade');
+
+// uncomment after placing your favicon in /public
+//app.use(favicon(__dirname + '/public/favicon.ico'));
+app.use(requestLogger());
+
+// The proxy must be set up before all the other middleware.
+// TODO: WE might want to move the middleware to each of the individual routes
+// so we don't have weird conflicts in the future.
+app.use('/elasticsearch', proxy);
+
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(cookieParser());
+app.use(compression());
+// app.use(require('less-middleware')(config.public_folder));
+app.use(express.static(config.public_folder));
+if (config.external_plugins_folder) app.use('/plugins', express.static(config.external_plugins_folder));
+
+app.use('/', routes);
+
+// catch 404 and forward to error handler
+app.use(function (req, res, next) {
+  var err = new Error('Not Found');
+  err.status = 404;
+  next(err);
+});
+
+// error handlers
+
+// development error handler
+// will print stacktrace
+if (app.get('env') === 'development') {
+  app.use(function (err, req, res, next) {
+    res.status(err.status || 500);
+    res.render('error', {
+      message: err.message,
+      error: err
+    });
+  });
+}
+
+// production error handler
+// no stacktraces leaked to user
+app.use(function (err, req, res, next) {
+  res.status(err.status || 500);
+  res.render('error', {
```

Comment:
Talked with @rashidkpc ... We are going to just send HTML for the errors. I'm gonna leave this code as is. We can add some styling to pretty them up down the road.